leshan-core/src/main/java/org/eclipse/leshan/senml/SenMLRecord.java

```diff
@@ -0,0 +1,208 @@
+/*******************************************************************************
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *     Boya Zhang - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.leshan.senml;
+
+import org.eclipse.leshan.core.model.ResourceModel;
+import org.eclipse.leshan.core.model.ResourceModel.Type;
+
+public class SenMLRecord {
+
+    private String name;
+    private Long time;
+
+    private Number floatValue;
+    private Boolean booleanValue;
+    private String objectLinkValue;
+    private String stringValue;
+    private Long timeValue;
+    private byte[] opaqueValue;
+
+    public ResourceModel.Type getType() {
+        if (booleanValue != null) {
+            return Type.BOOLEAN;
+        }
+        if (floatValue != null) {
+            return Type.FLOAT;
+        }
+        if (objectLinkValue != null) {
+            return Type.OBJLNK;
+        }
+        if (stringValue != null) {
+            return Type.STRING;
+        }
+        if (timeValue != null) {
+            return Type.TIME;
+        }
+        if (opaqueValue != null) {
+            return Type.OPAQUE;
+        }
+        return null;
+    }
+
+    public void setTime(Long time) {
+        this.time = time;
+    }
+
+    public Long getTime() {
+        return time;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public Long getTimeValue() {
+        return timeValue;
+    }
+
+    public void setTimeValue(Long time) {
```

Comment:
I didn't see any time value. 
Neither in https://tools.ietf.org/html/rfc8428#section-4.3 nor in http://www.openmobilealliance.org/release/LightweightM2M/V1_1-20180710-A/HTML-Version/OMA-TS-LightweightM2M_Core-V1_1-20180710-A.html#7-4-4-0-744-JSON